### ISSUE TITLE
ConcurrentModificationException

### ISSUE THREAD

**Issue by AndrejLukasevic** [intercom/intercom-android#119]
java.util.ConcurrentModificationException
       at java.util.ArrayList$ArrayListIterator.next(ArrayList.java:573)
       at io.intercom.android.nexus.NexusClient.fire(NexusClient.java:68)
       at io.intercom.android.nexus.NexusClient$1.run(NexusClient.java:119)
       at android.os.Handler.handleCallback(Handler.java:733)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:146)
       at android.os.HandlerThread.run(HandlerThread.java:61)

**Comment by harkin** [comment]
@AndrejLukasevic what version of the library are you using?

**Comment by jfsso** [comment]
We've been observing this on our app too. We're using SDK 1.1.3. The frequency is not high: so far 1 crash/user out of thousands of users.

```
java.util.ConcurrentModificationException
       at java.util.ArrayList$ArrayListIterator.next(ArrayList.java:573)
       at io.intercom.android.nexus.NexusClient.fire(SourceFile:68)
       at io.intercom.android.nexus.NexusClient$1.run(SourceFile:119)
       at android.os.Handler.handleCallback(Handler.java:739)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:145)
       at android.os.HandlerThread.run(HandlerThread.java:61)
```

@harkin I can also share the crashlytics issue with all threads dump with you in private. Let me know where I can send it.

**Comment by harkin** [comment]
Thanks @jfsso I think the stack traces here are enough to pinpoint what's going wrong. The library is trying to close a connection as it's also firing an event down that connection and it doesn't like that :( We'll  investigate why this is happening

**Comment by cazador4** [comment]
Same error here. SDK 1.0.0
The frecuency is not high also.

Thanks!

**Comment by harkin** [comment]
http://crashes.to/s/ae51e3f7b58 crash report
I'm looking into this one now

**Comment by DaleCantwell** [comment]
We fixed this issue in v1.1.5. Please contact us if you see any regressions.
